src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -402,13 +376,19 @@ async def wrapped(*args, **kwargs):
                 Tracer.push(trace)
                 enrich_span_with_input(span, trace.inputs)
                 output = await func(*args, **kwargs)
-                output = enrich_span_with_trace_type(span, trace.inputs, output, trace_type)
-                span.set_status(StatusCode.OK)
-                output = Tracer.pop(output)
+                if isinstance(output, AsyncIterator):
+                    output = Tracer.pop(output)
+                    output = traced_async_generator(span, trace.inputs, output, trace_type)
+                else:
+                    enrich_span_with_trace_type(span, trace.inputs, output, trace_type)
+                    span.set_status(StatusCode.OK)
+                    output = Tracer.pop(output)
             except Exception as e:
```

Comment:
Maybe we could have something like `should_end` signal to keep the logic more clean.

Reply:
Instead of using this signal, I'd like to wrap it in a function handle different output type.